### ISSUE TITLE
Custom keras optimizer - learning rate changes each epoch

### ISSUE THREAD

**Issue by rohanmohapatra** [keras-team/keras#13737]
I'm writing a custom optimizer where the learning rate changes on runtime for each epoch,  is there a way to pass changing `lr` to the Optimizer class on runtime?

Is there a way I can do it?
```
def get_updates(self, loss, params):
        grads = self.get_gradients(loss, params)
        self.updates = [K.update_add(self.iterations, 1)]
        
        lr = self.lr
        # momentum
        shapes = [K.int_shape(p) for p in params]
        moments = [K.zeros(shape) for shape in shapes]
        self.weights = [self.iterations] + moments
        for p, g, m in zip(params, grads, moments):
            v = self.momentum * m - lr * g  # velocity
            print(lr, g)
            self.updates.append(K.update(m, v))

        return self.updates
```

A custom l_r for each epoch?

**Comment by joshz123** [comment]
yes, you can use a learningratescheduler. https://github.com/keras-team/keras/issues/888 This issue has details about implementation. You can then use the scheduler to change the learning rate for each given epoch

**Comment by rohanmohapatra** [comment]
Got it thanks! :smile:
